index.js

```diff
@@ -157,8 +157,8 @@ HtmlWebpackPlugin.prototype.apply = function (compiler) {
       .then(function (html) {
         var pluginArgs = {html: html, assets: assets, plugin: self, outputName: self.childCompilationOutputName};
         return applyPluginsAsyncWaterfall('html-webpack-plugin-after-html-processing', pluginArgs)
-          .then(function () {
-            return pluginArgs.html;
+          .then(function (newPluginArgs) {
+            return newPluginArgs && newPluginArgs.html || pluginArgs.html;
```

Comment:
Shouldn't this be `newPluginArgs ? newPluginArgs.html : pluginArgs.html`?
Does this allow to be changed by more than one plugin?
Can you please add a unit test for this?